### PR TITLE
Add Code Cache Reclamation Doc

### PR DESCRIPTION
**Pull request by dsouzai**
Depends on https://github.com/eclipse/openj9/pull/7448

Closes https://github.com/eclipse/openj9/issues/1155

### CODE REVIEW

**Comment by dsouzai** [comment]
@mpirvu could take a look?

**Comment by dsouzai** [comment]
@mpirvu could you take a look at this when you have some time?

File: doc/compiler/runtime/CodeCacheReclamation.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Under these circumstances, the stale/invalid code takes up memory in the Code Cache 
+that could be used for something else.
+
+Code Cache Reclamation is the process by which unreachable code memory is reclaimed
+for reuse. The code generated for a method can either be reclaimed in a two-step 
+process, or all at once.
+
+# Two-Step Reclamation
+
+Code memory is reclaimed in two steps when the code is still reachable by executing
```
Comment (mpirvu) [inline doc/compiler/runtime/CodeCacheReclamation.md]: You may want to add that the two-step reclamation process is typically induced by recompilation.

File: doc/compiler/runtime/CodeCacheReclamation.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# Two-Step Reclamation
+
+Code memory is reclaimed in two steps when the code is still reachable by executing 
+threads, either because it is currently being executed, or because it is part of the 
+callstack(s). Under these circumstances, the code must remain available for all such 
+threads (though it will not be reachable via a new invocation). 
+
+Once it has been determined that the method body is stale, it is added to the Faint
+Cache Block list. However, not all of the stale body is slated to get reclaimed; the 
+pre-prologue and some number of instuctions are kept intact. This is to ensure that
```
Comment (mpirvu) [inline doc/compiler/runtime/CodeCacheReclamation.md]: Typo: instuctions

**Comment by dsouzai** [comment]
@mpirvu requested changes made.

**Comment by mpirvu** [comment]
Since this is just a documentation change, no testing is needed.

**Pull request merged**